tests/src/sanity.rs

```diff
@@ -0,0 +1,326 @@
+use std::{collections::HashMap, path::Path};
+
+use k8s_openapi::api::{batch::v1::Job, core::v1::Pod};
+use kube::{api::ListParams, Api};
+use reqwest::Method;
+use tokio::{
+    io::{AsyncBufReadExt, AsyncReadExt, BufReader},
+    time::{sleep, timeout, Duration},
+};
+
+use crate::utils::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[tokio::test]
+    // starts the node(express.js) api server, sends four different requests, validates data,
+    // stops the server and validates if the agent job and pod are deleted
+    async fn test_complete_node_api() {
+        let client = setup_kube_client().await;
+
+        let service_url = get_service_url(&client, "default").await.unwrap();
+        let pod_name = get_nginx_pod_name(&client, "default").await.unwrap();
+        let command = vec!["node", "node-e2e/app.js"];
+        let env: HashMap<&str, &str> = HashMap::new(); // for adding more environment variables
+        let mut server = start_node_server(&pod_name, command, env);
+
+        let mut stderr_reader = BufReader::new(server.stderr.take().unwrap());
+        let child_stdout = server.stdout.take().unwrap();
+
+        setup_panic_hook();
+
+        tokio::spawn(async move {
+            loop {
+                let mut error_stream = String::new();
+                // the following is a blocking call, so we need to spawn a task to read the stderr
+                stderr_reader.read_line(&mut error_stream).await.unwrap();
+                // Todo(investigate): it looks like the task reaches the panic when not kept in a
+                // loop i.e. the buffer reads an empty string:  `thread 'main'
+                // panicked at 'Error: '`
+                if !error_stream.is_empty() {
+                    panic!("Error: {}", error_stream);
+                }
+            }
+        });
+
+        // since we are reading from the stdout, we could block at any point in case the server
+        // does not write to its stdout, so we need a timeout here
+        let validation_duration = Duration::from_secs(30);
```

Comment:
I'd call this validation_timeout, not validation_duration. The validation duration should (hopefully) be a lot shorter.

Reply:
reduced the duration to 20 seconds